File: 002. Add Two Numbers/Java/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode fakeHead = new ListNode(0);
        ListNode temp = fakeHead;
        int carry = 0;
        while (l1 != null || l2 != null) {
            int current = carry;
            if (l1 != null) {
                current += l1.val;
                l1 = l1.next;
            }
            if (l2 != null) {
                current += l2.val;
                l2 = l2.next;
            }
            temp.next = new ListNode(current % 10);
            temp = temp.next;
            carry = current / 10;
        }
        if (carry != 0) {
            temp.next = new ListNode(carry);
        }
        return fakeHead.next;
    }
}

File: 002. Add Two Numbers/Python/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        fake_head = ListNode(0)
        temp = fake_head
        carry = 0
        while l1 or l2:
            current = carry
            if l1:
                current += l1.val
                l1 = l1.next
            if l2:
                current += l2.val
                l2 = l2.next
            temp.next = ListNode(current % 10)
            temp = temp.next
            carry = current // 10
        if carry:
            temp.next = ListNode(carry)
        return fake_head.next
